src/doc/rustc/src/codegen-options/index.md

```diff
@@ -39,42 +60,84 @@ metrics.
 ## lto
 
 This flag instructs LLVM to use [link time
-optimizations](https://llvm.org/docs/LinkTimeOptimization.html).
+optimizations](https://llvm.org/docs/LinkTimeOptimization.html) to produce
+better optimized code, using whole-program analysis, at the cost of longer
+linking time.
 
-It takes one of two values, `thin` and `fat`. 'thin' LTO [is a new feature of
-LLVM](http://blog.llvm.org/2016/06/thinlto-scalable-and-incremental-lto.html),
-'fat' referring to the classic version of LTO.
+This flag may take one of the following values:
+
+* `y`, `yes`, `on`, `fat`, or no value: Performs "fat" LTO which attempts to
+  perform optimizations across all crates within the dependency graph.
+* `n`, `no`, `off`: Disables LTO.
+* `thin`: Performs ["thin"
+  LTO](http://blog.llvm.org/2016/06/thinlto-scalable-and-incremental-lto.html).
+  This is similar to "fat", but takes substantially less time to run while
+  still achieving performance gains similar to "fat".
+
+If `-C lto` is not specified, then the compiler will attempt to perform "thin local LTO"
+which performs "thin" LTO on the local crate only across its [codegen
+units](#codegen-units). In this case, LTO is disabled if codegen units is 1 or
+optimizations are disabled ([`-C opt-level=0`](#opt-level)).
```

Comment:
Is this true? 
I tried to set exclusive lto and codegen-units option and the binary size is
increased than when I set both lto=true and codegen-units=1.

Reply:
By "exclusive LTO", do you mean not including the flag? Can you clarify which commands you ran?

I would expect it to be larger, since lto=true is "fat", and without the flag it is only "thin local", which does less optimization.  

[Here](https://github.com/rust-lang/rust/blob/f39205b5d9825fcf35989b5a04d115d411175d18/src/librustc/session/mod.rs#L607-L618) is the logic this is referring to.

Reply:
By "exclusive" I mean I either set lto or codegen-units.

Reply:
Also, I set these options in Cargo.toml, not via rustc arguments

Reply:
I'm still not sure I understand the confusion here.  `lto=true` + `codegen-units=1` should be the smallest.  It does fat LTO, and with 1 CGU per crate, so it has more optimization potential (but is very slow to compile).  That is:

- `codegen-units=1`: Disables LTO, so won't have cross-crate optimizations.
- `lto=true`: 16 cgus, performs fat LTO across crates.
- `codegen-units=1` + `lto=true`:  1 cgu, fat LTO across crates.

Reply:
IMO adding that to the documentation seems really helpful.

Reply:
Sure, I'll try to make it a little clearer.